### ISSUE TITLE
Sync fail on MacDesktop only

### ISSUE THREAD

**Issue by DblK** [dani-garcia/vaultwarden#4279]
### Subject of the issue
Syncing from the desktop MacOS Arm version from apple store does not sync.
However, mobile app and extension sync well.

### Deployment environment

* vaultwarden version: `1.30.1`

* Install method: Extracted today from docker using [this](https://github.com/dani-garcia/vaultwarden/wiki/Pre-built-binaries#extracting-binaries-without-docker-installed)

* Clients used: `MacOs Desktop M1 2023.12.1` (failing) & `Android 2024.1.1 (8843)` working & `Extension Chrome 2024.1.1` working

* Reverse proxy and version: `caddy v2.6.4 h1:2hwYqiRwk1tf3VruhMpLcYTg+11fCdr8S3jhNAdnPy8=`

* MySQL/MariaDB or PostgreSQL version: `sqlite`

* Other relevant details:

### Steps to reproduce

Nothing changed from default, just:
- launching the server
- connect to the self-hosted domain
- sync from desktop or android
- desktop fails whereas android is not

### Expected behaviour

Sync on desktop should not fail

### Actual behaviour

Desktop app is out of sync

### Troubleshooting data

Log from extension
```
[2024-01-28 00:20:15.116][request][INFO] POST /identity/connect/token
[2024-01-28 00:20:15.142][response][INFO] (login) POST /identity/connect/token => 200 OK
[2024-01-28 00:20:15.225][request][INFO] GET /api/sync
[2024-01-28 00:20:15.390][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
```

Log from desktop app
```
[2024-01-28 00:20:24.828][request][INFO] POST /identity/connect/token
[2024-01-28 00:20:24.852][response][INFO] (login) POST /identity/connect/token => 200 OK
[2024-01-28 00:20:24.918][request][INFO] GET /api/sync?excludeDomains=true
[2024-01-28 00:20:25.065][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
```

**Comment by BlackDex** [comment]
I'm not seeing errors in the logs.
Try to fully logout and back in, not lock, but logout.
That probably solves your issues.

**Comment by DblK** [comment]
Yes, I did what you suggested and `logout`/`login` was the right call. It's working again.
Thx
